01/Source.cpp

```diff
@@ -0,0 +1,176 @@
+#include <iostream>
+#include <exception>
+#include <algorithm>
+
+
+template <typename T>
+class Vector {
+	size_t _capacity;
+	size_t _size;
+	T* data;
+public:
+	Vector() :_capacity(13), _size(0), data(new T[_capacity]{}) {}
+	Vector(size_t arr_size) :_capacity(arr_size), _size(0), data(new T[_capacity]{}) {}
+	Vector(size_t arr_size, T elem) :_capacity(arr_size), _size(0), data(new T[_capacity]{ elem }) {}
+	Vector(const Vector<T>& other) :_capacity(other._capacity), _size(other._size), data(new T[_capacity]) {
+		for (int i = 0; i < _capacity; ++i) {
+			data[i] = other.data[i];
+		}
+
+	}
+	Vector(Vector<T>&& other) :Vector(other) {
+		other.data = NULL;
+	}
+
+	~Vector() {
+		delete[] data;
+	}
+
+	Vector<T>& operator=(const Vector<T>& other) {
+		if (this == other) {
```

Comment:
`this` is a pointer, `other` is not, you should compare `this` with the address of `other`